### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Preadditive.HomOrthogonal

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      have : z ≫ biproduct.π (fun b ↦ s (g b)) b ≫ biproduct.ι (fun b ↦ s (g b)) b ≫ w ≫
+          biproduct.π (fun b ↦ s (h b)) c = 0 := by
+        have : biproduct.π (fun b ↦ s (g b)) b ≫ biproduct.ι (fun b ↦ s (g b)) b ≫ w ≫
+            biproduct.π (fun b ↦ s (h b)) c = 0 := by
+          have : biproduct.ι (fun b ↦ s (g b)) b ≫ w ≫ biproduct.π (fun b ↦ s (h b)) c = 0 := by
+            apply o.eq_zero nm
+          rw [this, comp_zero]
+        rw [this, comp_zero]
+      rw [this, comp_zero]
+    rw [this, comp_zero]
```
Comment (mo271) [inline Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:182]: needs some golfing

Comment (mo271) [reply Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:182]: I tried `refine' comp_zero`, even providing some of the arguments, but no luck. Also `convert` does not seem up for the task anymore, even when I try to get more explicit with `convert @comp_zero ...`

Comment (mo271) [reply Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:182]: perhaps it is fine to leave it as is with the fix, accepting that `convert` is now a little weaker...

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean [hunk 2]
```diff
@@ -130,47 +134,58 @@ noncomputable def matrixDecompositionAddEquiv (o : HomOrthogonal s) {α β : Typ
...
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:181]: Untested, but would this also work?
```suggestion
    have : biproduct.ι (fun b ↦ s (g b)) b ≫ w ≫ biproduct.π (fun b ↦ s (h b)) c = 0 := by
      apply o.eq_zero nm
    simp only [this, comp_zero]
```

Comment (mo271) [reply Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean]: seems to work, great!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**